app/waf/src/controllers/adc.controller.ts

```diff
@@ -887,16 +879,9 @@ export class AdcStateCtrlr {
   private async partitioned(ctrl: AdcStateCtrlr): Promise<boolean> {
     let bigipMgr = await ctrl.getBigipMgr();
     let partition = as3Name(ctrl.adc.tenantId);
-    try {
-      let resObj = await bigipMgr.getPartition(partition);
-      let code = JSON.parse(resObj)['body'][0]['name'];
-      if (code !== partition) {
-        return false;
-      }
-    } catch (error) {
-      return false;
-    }
-    return true;
+    let resObj = await bigipMgr.getPartition(partition);
+    let code = JSON.parse(resObj)['body'][0]['name'];
+    return code === partition;
```

Comment:
do we need to try to catch up the exception?  the getPartition function might throw some exceptions?

Reply:
To the functions `partitioned` `installed` ...(these check functions), we only expect 2 results: `true` and `non-true` for now. 
We don't care `non-true` is either false or exception because the outside caller can help us to handle the exception. 
So we need not catch the exception here to make the process simpler.

Reply:
ok. get it. I am ok if we definitely don't care about the "non-true" case.